R/filter-dm.R

```diff
@@ -89,13 +91,16 @@ set_filter_for_table <- function(dm, table, filter_exprs, zoomed) {
 #'   cdm_filter(planes, engine %in% c("Reciprocating", "4 Cycle")) %>%
 #'   compute()
 #' @export
-cdm_apply_filters <- function(dm) {
+cdm_apply_filters <- function(dm, ...) {
   check_not_zoomed(dm)
-  def <- cdm_get_def(dm)
-
-  def$data <- map(def$table, ~ tbl(dm, .))
-
-  cdm_reset_all_filters(new_dm3(def))
+  vars <- tidyselect_table_names(dm)
+  selected <- tidyselect::vars_select(vars, ...)
+  # in case of empty ellipsis all tables should be in selection
+  if (is_empty(selected)) selected <- names(dm)
+  new_def <- cdm_get_def(dm) %>%
+    mutate(data = map(table, ~if_else(.x %in% selected, list(tbl(dm, .x)), data[.x == table])) %>% flatten)
+
+  cdm_reset_all_filters(new_dm3(new_def))
```

Comment:
Is this still correct?

Reply:
Very valid point. In my opinion, it should be still correct and we should add something like the following to the documentation: "A call to `cdm_apply_filters()` resets all filter conditions that are stored in the `dm`, independent if the filters are only applied to a subset of the tables"

But I'm open to suggestions

Reply:
I forgot about the duality here. In this case let's shelve this for now until we have a better understanding of filtering and applying filters.

Reply:
It somehow doesn't seem very likely to me, that we're going to move away far from our current API regarding the filtering, apart from #184, which doesn't affect this issue though. I might be wrong, but it seems to me that it's just a question of deciding whether we want the filter conditions still to be stored or dropped after `cdm_apply_filters()`